### ISSUE TITLE
TypeScript: Comma as separator for single-line object literal type

### ISSUE THREAD

**Issue by KSXGitHub** [dprint/dprint#202]
**Is your feature request related to a problem? Please describe.**

I want to write this code:

```typescript
type MyObject = { foo: number, bar: string }
```

But dprint-plugin-typescript converts it to semicolons:

```typescript
type MayObject = { foo: number; bar: string }
```

which it is not what I want.

**Describe the solution you'd like**

While I prefer commas as separator, others might prefer semicolons, so I suggest a configuration key should be introduced to specify whether the separator should be commas or semicolons:
* `objectLiteralTypeSeparator: 'comma' | 'semicolon'`
* `singleLineObjectLiteralTypeSeparator: 'comma' | 'semicolon''`
* `multiLineObjectLiteralTypeSeparator: 'comma' | 'semicolon' | 'none'`

**Describe alternatives you've considered**

**Additional context**

I have set `.semi_colons(SemiColons::Asi)` in [my own formatter](https://github.com/KSXGitHub/sane-fmt/blob/0.2.16/src/rules/mod.rs#L14), so I would prefer multi-line object literal type to **not** have separator.

**Comment by dsherret** [comment]
@KSXGitHub yes, that would be a good idea! Probably just `"typeLiteral.use_comma_separators": true`.

By the way, have you see that dprint already has a rust cli? https://crates.io/crates/dprint

Also, I've been working on #194 for plugins in the rust cli. I think in the future you would just have to do `dprint --config https://yourdomain.com/path/to/your/opinionated/config.json`

**Comment by KSXGitHub** [comment]
@dsherret 

> By the way, have you seen that dprint already has a rust cli? https://crates.io/crates/dprint

1. You did not provide a way to easily install dprint without installing either Node.js or Rust. One of the reasons I created sane-fmt was because I to want to easily install it in CI environment without requiring another toolchain. To fix this, you should release prebuilt binaries to GitHub Release <sub>(I use GitHub Actions CI to automatically upload prebuilt binaries to GitHub Release, publish NPM packages, and update AUR packages, you might want to check out).</sub>

2. It still requires a config file: While I don't mind copying my config file to a new repo, keeping them up-to-date would mean copying back and forth. I am way too lazy for that.

3. It is easier to add my own features.
